src/Property.php

```diff
@@ -79,10 +82,23 @@ public function isNullable(): bool
 
     protected function resolveTypeDefinition()
     {
+        if (isset(self::$cache[$this->class][$this->getName()])) {
+            $this->isNullable = self::$cache[$this->class][$this->getName()]['is_nullable'];
+            $this->hasTypeDeclaration = (
+                ! empty(self::$cache[$this->class][$this->getName()]['types'])
+                || ! empty(self::$cache[$this->class][$this->getName()]['array_types'])
+            );
+            $this->types = self::$cache[$this->class][$this->getName()]['types'] ?? [];
+            $this->arrayTypes = self::$cache[$this->class][$this->getName()]['array_types'] ?? [];
+
+            return;
```

Comment:
Move this to a separate protected method `resolveFromCache` or `restoreFromCache`, maybe even another name?